third-culture-software/bhima
# Filters for required inventory scans registry

Added custom filters to Required Inventory Scans registry
- reference number
- period (for end date)

Closes:  https://github.com/IMA-WorldHealth/bhima/issues/6480

Also registry action menu item in the Required Inventory Scans registry to go to the Assets Scans Page and show the scans that satisfy this required inventory.   This is mentioned as part of [Issue 6481](https://github.com/IMA-WorldHealth/bhima/issues/6481) but does not close it.

Changes the default period for the Asset Scans registry and the Required Inventory Scans registry to 'this year'.

**TESTING**
- Use bhima_test
- In Asset Management > Required Inventory Scans
   - Create several new scans with a variety of reference_number and end_dates
   - Use the custom filters to test them
   - Use the action item "Show scanned assets" item to test that function.   Note that the filtering involved handles the period (start of inventory scan to end of inventory scan) as well as the optional Depot.  Test various combinations.

**NOTE:** I had some problems with the Required Inventory Scans registry period default filter (for the end date).  But it seemed to work okay for Chris.   And it seems to work since I did more work on the PR.  But I encourage testing it thoroughly.
